app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/drawer/MainActivityDrawer.java

```diff
@@ -74,13 +75,24 @@ private void setupActionBar() {
   }
 
   private void initViewModel() {
-    mViewModel = mDrawerViewModelFactory.create(mMainActivity, new AndroidTreeViewNodeDisplayer());
+    mViewModel = mDrawerViewModelFactory.create(mMainActivity, new AndroidTreeViewNodeDisplayer(),
+        this::openNewDynamicLayerDialog);
     bindViewModel();
     mViewModel.setView(this);
     mViewModel.start();
     mRecyclerView.setAdapter(mViewModel.getUsersAdapter());
   }
 
+  private void openNewDynamicLayerDialog() {
+    new EditTextAlertDialogBuilder(mMainActivity).setIsCancelable(true)
+        .setTextValidator(mViewModel.getDynamicLayerTextValidator())
```

Comment:
Isn't this weird, this somehow cyclic reference that the `ViewModel` will hold to itself? I know it is absolutely OK, but I admit it took me a moment to figure it out